File: samples/AdventureWorksLT.WpfApp/PercentageConverter.cs
using System;
using System.Globalization;
using System.Windows;
using System.Windows.Data;

namespace DevZest.Samples.AdventureWorksLT
{
    public sealed class PercentageConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (targetType == null || targetType != typeof(string))
                return DependencyProperty.UnsetValue;

            if (value == null)
                return null;

            if (value is decimal?)
                return string.Format("{0:P}", value);

            return DependencyProperty.UnsetValue;
        }

        private bool TryConvertBack(string value, CultureInfo culture, out decimal? result)
        {
            result = new decimal?();
            if (string.IsNullOrEmpty(value))
                return true;

            if (TryConvertBack(value, culture, out decimal decimalResult))
            {
                result = decimalResult;
                return true;
            }

            return false;
        }

        private bool TryConvertBack(string value, CultureInfo culture, out decimal result)
        {
            result = default(decimal);
            if (string.IsNullOrEmpty(value))
                return false;

            try
            {
                var text = value.Trim();
                if (!culture.IsNeutralCulture && text.Length > 0 && culture.NumberFormat != null)
                {
                    switch (culture.NumberFormat.PercentPositivePattern)
                    {
                        case 0:
                        case 1:
                            if (text.Length - 1 == text.LastIndexOf(culture.NumberFormat.PercentSymbol, StringComparison.CurrentCultureIgnoreCase))
                                text = text.Substring(0, text.Length - 1);
                            break;
                        case 2:
                            if (text.IndexOf(culture.NumberFormat.PercentSymbol, StringComparison.CurrentCultureIgnoreCase) == 0)
                                text = text.Substring(1);
                            break;
                    }
                }
                result = System.Convert.ToDecimal(text, culture);
                return true;
            }
            catch (ArgumentOutOfRangeException)
            {
            }
            catch (ArgumentNullException)
            {
            }
            catch (FormatException)
            {
            }
            catch (OverflowException)
            {
            }
            return false;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            string text;
            if (value == null)
                text = null;
            else if (value is string)
                text = (string)value;
            else
                return DependencyProperty.UnsetValue;

            if (targetType == typeof(decimal?))
                return TryConvertBack(text, culture, out decimal? result) ? result / 100 : DependencyProperty.UnsetValue;
            else
                return DependencyProperty.UnsetValue;
        }
    }
}

File: samples.vb/AdventureWorksLT.WpfApp/PercentageConverter.vb
Imports System.Globalization

Public NotInheritable Class PercentageConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

        If targetType Is Nothing OrElse targetType <> GetType(String) Then
            Return DependencyProperty.UnsetValue
        End If

        If value Is Nothing Then
            Return Nothing
        End If

        If TypeOf value Is System.Nullable(Of Decimal) Then
            Return String.Format("{0:P}", value)
        End If

        Return DependencyProperty.UnsetValue
    End Function

    Private Function TryConvertBack(value As String, culture As CultureInfo, ByRef result As Decimal?) As Boolean
        result = New Decimal?()
        If String.IsNullOrEmpty(value) Then
            Return True
        End If

        Dim decimalResult As Decimal = 0
        If TryConvertBack(value, culture, decimalResult) Then
            result = decimalResult
            Return True
        End If
        Return False
    End Function

    Private Function TryConvertBack(value As String, culture As CultureInfo, ByRef result As Decimal) As Boolean
        result = Nothing
        If String.IsNullOrEmpty(value) Then
            Return False
        End If

        Try
            Dim text = value.Trim()
            If Not culture.IsNeutralCulture AndAlso text.Length > 0 AndAlso culture.NumberFormat IsNot Nothing Then
                Select Case culture.NumberFormat.PercentPositivePattern
                    Case 0, 1
                        If text.Length - 1 = text.LastIndexOf(culture.NumberFormat.PercentSymbol, StringComparison.CurrentCultureIgnoreCase) Then
                            text = text.Substring(0, text.Length - 1)
                        End If
                        Exit Select
                    Case 2
                        If text.IndexOf(culture.NumberFormat.PercentSymbol, StringComparison.CurrentCultureIgnoreCase) = 0 Then
                            text = text.Substring(1)
                        End If
                        Exit Select
                End Select
            End If
            result = System.Convert.ToDecimal(text, culture)
            Return True
        Catch ex As ArgumentOutOfRangeException
        Catch ex As ArgumentNullException
        Catch ex As FormatException
        Catch ex As OverflowException
        End Try
        Return False
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        Dim text As String
        If value Is Nothing Then
            text = Nothing
        ElseIf TypeOf value Is String Then
            text = CType(value, String)
        Else
            Return DependencyProperty.UnsetValue
        End If

        If targetType Is GetType(Decimal?) Then
            Dim result As Decimal? = Nothing
            Return If(TryConvertBack(text, culture, result), result / 100, DependencyProperty.UnsetValue)
        Else
            Return DependencyProperty.UnsetValue
        End If
    End Function
End Class